osu.Game.Tournament/Screens/Gameplay/GameplayScreen.cs

```diff
@@ -152,6 +189,58 @@ protected override void CurrentMatchChanged(ValueChangedEvent<TournamentMatch> m
             scheduledOperation?.Cancel();
         }
 
+        private void beatmapChanged(ValueChangedEvent<BeatmapInfo> beatmap)
+        {
+            if (beatmap.NewValue == null || CurrentMatch.Value == null)
+                return;
+
+            var currentSelectedBeatmapMods = CurrentMatch.Value.Round.Value.Beatmaps.FirstOrDefault(b => b.ID == beatmap.NewValue.OnlineBeatmapID)?.Mods;
+            showVideoForModsAndHideAllElse(currentSelectedBeatmapMods);
+        }
+
+        private ScheduledDelegate hideVideosScheduleOperation;
+
+        private void showVideoForModsAndHideAllElse(string currentSelectedBeatmapMods)
+        {
+            hideVideosScheduleOperation?.Cancel();
+            var videosThatShouldBeHidden = new List<TourneyVideo>();
+            TourneyVideo videoToShow = null;
+
+            foreach (var (mods, video) in modSpecificGameplayVideos)
+            {
+                if (mods.Equals(currentSelectedBeatmapMods, StringComparison.OrdinalIgnoreCase))
+                {
+                    videoToShow = video;
+                }
+                else
+                {
+                    videosThatShouldBeHidden.Add(video);
+                }
+            }
+
+            if (videoToShow == null)
+            {
+                videoToShow = fallbackGameplayVideo;
+            }
+            else
+            {
+                videosThatShouldBeHidden.Add(fallbackGameplayVideo);
+            }
+
+            // put video to show on top, and fade it in
+            videosContainer.ChangeChildDepth(videoToShow, (float)-Clock.CurrentTime);
+            videoToShow.FadeIn(FADE_DELAY);
+
+            // after the fade, hide all others
+            hideVideosScheduleOperation = Scheduler.AddDelayed(() =>
```

Comment:
You can use `.OnComplete()` chained to the transform to avoid needing to track/cancel/delay this.

Reply:
I started playing with it a bit but my first implementation where I got rid of the Scheduler call and put the hide logic in the OnComplete of the FadeIn transform I was able to get weird results if I give it a second map update event before the fade completed. Sometimes it would result in wrong video being shown, I had one case in testing where everything ended up hidden (not sure how I got that, might have been some other issue), anyway, it clearly was doing not the right thing in that scenario (two changes quickly followed after each other).

I think some sort of cancel logic is still needed, I didnt have enough time to look into exactly how the transform stuff works and if its possible to do in there, and I likely wont have much time this week either as I'll be getting pretty busy with work... So I'm afraid I probably wont be able to update this till the weekend

Reply:
When you get to this, please also move this whole code into its own component (there's no reason for it to exist in `GameplayScreen`, and it's a sizeable block of code).

As for the issue you encountered, it would be easier to look into if there was test coverage for this. Will wait for code to be posted (feel free to post code in the mean time if you'd like a quick sanity check).